File: generators/themeParkNames.py
__all__ = ['themeParkNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm2', 'nm3', 'nm4', 'nameGen'])
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['result'])
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            if (var.get('i')<Js(5.0)):
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                var.put('names', (var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2'))))
            else:
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                var.put('names', ((var.get('nm3').get(var.get('rnd'))+Js(' '))+var.get('nm4').get(var.get('rnd2'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js('Angel'), Js('Animal'), Js('Aqua'), Js('Arcane'), Js('Astral'), Js('Astro'), Js('Aura'), Js('Beach'), Js('Beast'), Js('Carny'), Js('Cartoon'), Js('Child'), Js('Clown'), Js('Comic'), Js('Creep'), Js('Critter'), Js('Crypt'), Js('Demon'), Js('Dino'), Js('Doll'), Js('Dragon'), Js('Dread'), Js('Dream'), Js('Elf'), Js('Ember'), Js('Epic'), Js('Eterni'), Js('Ever'), Js('Expo'), Js('Fable'), Js('Fairy'), Js('Feral'), Js('Festi'), Js('Film'), Js('Fire'), Js('Forest'), Js('Freak'), Js('Fright'), Js('Fun'), Js('Game'), Js('Ghost'), Js('Giant'), Js('Groovy'), Js('Happy'), Js('Hell'), Js('Hero'), Js('Horror'), Js('Ice'), Js('Jungle'), Js('Kids'), Js('Luna'), Js('Lunar'), Js('Magic'), Js('Marina'), Js('Maze'), Js('Mega'), Js('Mini'), Js('Miracle'), Js('Mirror'), Js('Monster'), Js('Movie'), Js('Mutant'), Js('Never'), Js('Night'), Js('Ocean'), Js('Paradox'), Js('Phantom'), Js('Play'), Js('Quest'), Js('Rain'), Js('Rainbow'), Js('River'), Js('Robot'), Js('Saga'), Js('Sand'), Js('Scream'), Js('Secret'), Js('Shadow'), Js('Shock'), Js('Sky'), Js('Snow'), Js('Solar'), Js('Space'), Js('Speed'), Js('Spirit'), Js('Splash'), Js('Star'), Js('Stellar'), Js('Storm'), Js('Story'), Js('Summer'), Js('Sun'), Js('Super'), Js('Terra'), Js('Terror'), Js('Thrill'), Js('Titan'), Js('Toy'), Js('Undead'), Js('Vision'), Js('Warp'), Js('Water'), Js('Winter'), Js('Witch'), Js('Wizard'), Js('Wonder'), Js('Zombie'), Js('Zoo')]))
var.put('nm2', Js([Js('land'), Js('world'), Js('zone'), Js('park'), Js('town'), Js('fair'), Js('realm'), Js('ville'), Js('land'), Js('park'), Js('ventures')]))
var.put('nm3', Js([Js('Adventure'), Js('Amazing'), Js('Amazon'), Js('Angel'), Js('Animal'), Js('Animation'), Js('Anime'), Js('Aqua'), Js('Arcane'), Js('Astral'), Js('Beach'), Js('Beast'), Js('Bird'), Js('Candy'), Js('Cartoon'), Js('Castle'), Js('Child'), Js('Chocolate'), Js('Clown'), Js('Comic'), Js('Creep'), Js('Critter'), Js('Crypt'), Js('Crystal'), Js('Demon'), Js('Dino'), Js('Dinosaur'), Js('Discovery'), Js('Doll'), Js('Dragon'), Js('Dread'), Js('Dream'), Js('Elf'), Js('Ember'), Js('Enchanted'), Js('Epic'), Js('Fable'), Js('Fairy'), Js('Fairy Tale'), Js('Family'), Js('Fantasy'), Js('Feral'), Js('Festival'), Js('Film'), Js('Fire'), Js('Forest'), Js('Freak'), Js('Fright'), Js('Fun'), Js('Galaxy'), Js('Game'), Js('Ghost'), Js('Giant'), Js('Happy'), Js('Hell'), Js('Hero'), Js('Horror'), Js('Ice'), Js('Jungle'), Js('Kids'), Js("King's"), Js('Knight'), Js('Legend'), Js('Luna'), Js('Lunar'), Js('Magic'), Js('Magic Forest'), Js('Marina'), Js('Marine'), Js('Maze'), Js('Mega'), Js('Midnight'), Js('Mini'), Js('Miniature'), Js('Miracle'), Js('Mirror'), Js('Monster'), Js('Movie'), Js('Mutant'), Js('Mystery'), Js('Mystic'), Js('Mythic'), Js('Nature'), Js('Night'), Js('Nightmare'), Js('Oasis'), Js('Ocean'), Js('Panorama'), Js('Paradox'), Js('Parody'), Js('Phantom'), Js('Phoenix'), Js('Pirate'), Js('Play'), Js('Power'), Js('Quest'), Js('Rain'), Js('Rainbow'), Js('River'), Js('Robot'), Js('Saga'), Js('Sand'), Js("Santa's"), Js('Science'), Js('Scream'), Js('Secret'), Js('Serpent'), Js('Shadow'), Js('Shock'), Js('Sky'), Js('Snow'), Js('Solar'), Js('Space'), Js('Speed'), Js('Spirit'), Js('Splash'), Js('Star'), Js('Storebook'), Js('Storm'), Js('Story'), Js('Summer'), Js('Sun'), Js('Super'), Js('Surprise'), Js('Terror'), Js('Thrill'), Js('Titan'), Js('Toy'), Js('Trampoline'), Js('Transilvanian'), Js('Undead'), Js('Underwater'), Js('Universe'), Js('Vampire'), Js('Vision'), Js('Warp'), Js('Water'), Js('Werewolf'), Js('Wild Water'), Js('Wild West'), Js('Wildlife'), Js('Winter'), Js('Witch'), Js('Wizard'), Js('Wonder'), Js('Zombie'), Js('Zoo')]))
var.put('nm4', Js([Js('World'), Js('Land'), Js('Zone'), Js('Park'), Js('Town'), Js('Village'), Js('Realm'), Js('Fair'), Js('Island'), Js('Fun Park'), Js('Fun World'), Js('Kingdom'), Js('Dome'), Js('Paradise'), Js('Experience')]))
pass
pass


# Add lib to the module scope
themeParkNames = var.to_python()

File: js/themeParkNames.js
var nm1 = ["Angel","Animal","Aqua","Arcane","Astral","Astro","Aura","Beach","Beast","Carny","Cartoon","Child","Clown","Comic","Creep","Critter","Crypt","Demon","Dino","Doll","Dragon","Dread","Dream","Elf","Ember","Epic","Eterni","Ever","Expo","Fable","Fairy","Feral","Festi","Film","Fire","Forest","Freak","Fright","Fun","Game","Ghost","Giant","Groovy","Happy","Hell","Hero","Horror","Ice","Jungle","Kids","Luna","Lunar","Magic","Marina","Maze","Mega","Mini","Miracle","Mirror","Monster","Movie","Mutant","Never","Night","Ocean","Paradox","Phantom","Play","Quest","Rain","Rainbow","River","Robot","Saga","Sand","Scream","Secret","Shadow","Shock","Sky","Snow","Solar","Space","Speed","Spirit","Splash","Star","Stellar","Storm","Story","Summer","Sun","Super","Terra","Terror","Thrill","Titan","Toy","Undead","Vision","Warp","Water","Winter","Witch","Wizard","Wonder","Zombie","Zoo"];
var nm2 = ["land","world","zone","park","town","fair","realm","ville","land","park","ventures"];
var nm3 = ["Adventure","Amazing","Amazon","Angel","Animal","Animation","Anime","Aqua","Arcane","Astral","Beach","Beast","Bird","Candy","Cartoon","Castle","Child","Chocolate","Clown","Comic","Creep","Critter","Crypt","Crystal","Demon","Dino","Dinosaur","Discovery","Doll","Dragon","Dread","Dream","Elf","Ember","Enchanted","Epic","Fable","Fairy","Fairy Tale","Family","Fantasy","Feral","Festival","Film","Fire","Forest","Freak","Fright","Fun","Galaxy","Game","Ghost","Giant","Happy","Hell","Hero","Horror","Ice","Jungle","Kids","King's","Knight","Legend","Luna","Lunar","Magic","Magic Forest","Marina","Marine","Maze","Mega","Midnight","Mini","Miniature","Miracle","Mirror","Monster","Movie","Mutant","Mystery","Mystic","Mythic","Nature","Night","Nightmare","Oasis","Ocean","Panorama","Paradox","Parody","Phantom","Phoenix","Pirate","Play","Power","Quest","Rain","Rainbow","River","Robot","Saga","Sand","Santa's","Science","Scream","Secret","Serpent","Shadow","Shock","Sky","Snow","Solar","Space","Speed","Spirit","Splash","Star","Storebook","Storm","Story","Summer","Sun","Super","Surprise","Terror","Thrill","Titan","Toy","Trampoline","Transilvanian","Undead","Underwater","Universe","Vampire","Vision","Warp","Water","Werewolf","Wild Water","Wild West","Wildlife","Winter","Witch","Wizard","Wonder","Zombie","Zoo"];
var nm4 = ["World","Land","Zone","Park","Town","Village","Realm","Fair","Island","Fun Park","Fun World","Kingdom","Dome","Paradise","Experience"];
var br = "";

function nameGen(){
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		if(i < 5){
			rnd = Math.floor(Math.random() * nm1.length);
			rnd2 = Math.floor(Math.random() * nm2.length);
			names = nm1[rnd] + nm2[rnd2];
		}else{
			rnd = Math.floor(Math.random() * nm3.length);
			rnd2 = Math.floor(Math.random() * nm4.length);
			names = nm3[rnd] + " " + nm4[rnd2];
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}